### PR TITLE
Improve error message for invalid API result

### PR DESCRIPTION
**Pull request by SimonRichardson**
It is expected that we can always get valid API results, except when we
don't we should return an error message that is more informative. In
this instance, we end up returning an error message about missing bases.

The following code just improves that by giving a bit more context. I
can't change the error type as we use that to diagnose other issues, but
I do wish that `not found` wouldn't be appended to the end of the error
message as it's rather restricting on the message returned.

## QA steps

Bootstrap to microk8s

```sh
$ juju deploy ambassador
ERROR refresh: no bases located for architecture "amd64" not found
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1925728

### CODE REVIEW

File: apiserver/facades/client/charms/charmhubrepo.go [hunk 1]
```diff
@@ -271,12 +271,12 @@ func (c *chRepo) selectNextBase(bases []transport.Base, origin corecharm.Origin)
 		break
 	}
 	if !found {
-		return corecharm.Platform{}, errors.NotFoundf("base")
+		return corecharm.Platform{}, errors.NotFoundf("no bases located for architecture %q", origin.Platform.Architecture)
```
Comment (manadart) [inline apiserver/facades/client/charms/charmhubrepo.go:274]:
```suggestion
		return corecharm.Platform{}, errors.NotFoundf("bases matching architecture %q", origin.Platform.Architecture)
```
Then it reads better

**Review by manadart — CHANGES_REQUESTED** [review]
We can work with the " not found" appendage better here.

**Review by manadart — APPROVED** [review]
Nice.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**